apis/management.cattle.io/v3/rke_types.go

```diff
@@ -59,6 +59,14 @@ type RancherKubernetesEngineConfig struct {
 	RotateCertificates *RotateCertificates `yaml:"rotate_certificates,omitempty" json:"rotateCertificates,omitempty"`
 	// DNS Config
 	DNS *DNSConfig `yaml:"dns" json:"dns,omitempty"`
+	// Upgrade Strategy for the cluster
+	UpgradeStrategy *NodeUpgradeStrategy `yaml:"upgrade_strategy,omitempty" json:"upgradeStrategy,omitempty"`
+}
+
+type NodeUpgradeStrategy struct {
+	MaxUnavailable string         `yaml:"max_unavailable" json:"maxUnavailable,omitempty" norman:"default=50%"`
```

Comment:
Is there no benefit in having this as int?

Reply:
I'm accepting a string so that a number and a percentage both can be passed. It gets parsed by the k8s intstr package https://github.com/rancher/rke/pull/1800/files#diff-ee57d1a642f3b4ee28fb4929afe13026R64